crates/sargon/src/profile/logic/create_entity.rs

```diff
@@ -0,0 +1,71 @@
+use crate::prelude::*;
+
+impl Profile {
+    /// Creates `count` many new virtual entities of type `E` on `network_id` with `factor_source` as the factor source.
+    /// Setting the names according to `get_name`, loading pre-derived FactorInstances from the
+    /// FactorInstancesCache if possible, else derives more using the FactorInstancesProvider.
+    ///
+    /// Returns the FactorSourceID, the entities, the InstancesInCacheConsumer, and the FactorInstancesProviderOutcomeForFactor.
+    ///
+    /// The `FactorInstancesProviderOutcomeForFactor` is primarily useful for testing.
+    ///
+    /// The `InstancesInCacheConsumer` SHOULD be called by the caller, once you know it
+    /// is safe to delete the instances from the cache - e.g. after having saved the new
+    /// entities into the Profile and persisted it into SecureStorage.
+    pub async fn create_unsaved_entities_with_factor_source_with_derivation_outcome<
+        E: IsEntity + Identifiable,
+    >(
+        &self,
+        factor_source: FactorSource,
+        network_id: NetworkID,
+        count: u16,
+        factor_instances_cache_client: Arc<FactorInstancesCacheClient>,
+        key_derivation_interactors: Arc<dyn KeysDerivationInteractors>,
+        get_name: impl Fn(u32) -> DisplayName, // name of entity at index
+    ) -> Result<(
+        FactorSourceID,
+        IdentifiedVecOf<E>,
+        InstancesInCacheConsumer,
+        FactorInstancesProviderOutcomeForFactor,
+    )> {
+        let count = count as usize;
+
+        let fsid = factor_source.factor_source_id();
+
+        let (instances_in_cache_consumer, outcome) =
+            VirtualEntityCreatingInstanceProvider::for_many_entity_vecis(
+                count,
+                E::entity_kind(),
+                factor_instances_cache_client,
+                Arc::new(self.clone()),
+                factor_source,
+                network_id,
+                key_derivation_interactors,
+            )
+            .await?;
+
+        let instances_to_use_directly = outcome.clone().to_use_directly;
+
+        assert_eq!(instances_to_use_directly.len(), count);
```

Comment:
I understand that we need `count` instances, but the docs of `FactorInstancesProviderOutcomeForFactor.to_use_directly` state:
> FactorInstances which are not saved into the cache.

so that's why I asked why none can be saved on cache. Maybe that comment is wrong?

Reply:
no I think everything is correct, I think you might have misunderstood something perhaps? So this is how this works:

if we need 5 instances, as in `VirtualEntityCreatingInstanceProvider` is called with `for_many_entity_vecis(5)` we are going to use 5 to create 5 accounts. Those 5 instances should never be saved into the cache. They might all have been **read** from the cache, or 1 of them or 4 of them might have been read from the cache. If the cache did not contain 5, lets say it contains 1, then 4 will have been derived. 1+4 is "to be used" - but since we needed to derive **anyway** the FIP is coded to ALSO derive 30 more. So 34 will be derived. 1+4 to be used directly (create accounts), 30 instances to be cached.

Does that make sense?